src/Framework/Constraint/ArraySubset.php

```diff
@@ -130,4 +128,83 @@ private function toArray(iterable $other): array
         // Keep BC even if we know that array would not be the expected one
         return (array) $other;
     }
+
+    private function isAssociative(array $array): bool
+    {
+        return \array_reduce(\array_keys($array), function (bool $carry, $key): bool {
+            return $carry || \is_string($key);
+        }, false);
+    }
+
+    private function compare($first, $second): bool
+    {
+        return $this->strict ? $first === $second : $first == $second;
+    }
+
+    private function deepSort(array &$array): void
+    {
+        foreach ($array as &$value) {
+            if (\is_array($value)) {
+                $this->deepSort($value);
+            }
+        }
+
+        if ($this->isAssociative($array)) {
+            \ksort($array);
```

Comment:
This is weird. Are you sure this shouldn't be `asort`?

Reply:
The goal is to be able to compare the two arrays, and so both `ksort()` and `asort()` do the trick for this use case.

Reply:
Got it, as long as they end up in the same order, you are good. Thanks for the quick reply.